src/System.Runtime/tests/System/GCTests.netstandard1.7.cs

```diff
@@ -0,0 +1,196 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Threading;
+using System.Runtime;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Tests
+{
+    public class GCExtendedTests : RemoteExecutorTestBase
+    {
+        private const long MaxGarbage = 1000;
+        public static void MakeSomeGarbage()
+        {
+            Version vt;
+
+            for (int i = 0; i < MaxGarbage; i++)
+            {
+                // Create objects and release them to fill up memory
+                // with unused objects.
+                vt = new Version();
+            }
+        }
+
+        [Fact]
+        public static void GetGeneration_WeakReference()
+        {
+
+            Func<WeakReference> getweakref = delegate ()
+            {
+                Version myobj = new Version();
+                var wkref = new WeakReference(myobj);
+                Assert.True(GC.GetGeneration(wkref) >= 0);
+                Assert.Equal(GC.GetGeneration(wkref), GC.GetGeneration(myobj));
```

Comment:
This is flaky test - if GC kicks between the two GetGeneration calls, the returned values will be different.